src/operations/query/r4.js

```diff
@@ -101,142 +101,82 @@ class R4SearchQueryCreator {
          * this is used to pick index hints
          * @type {Set}
          */
-        let columns = new Set();
+        let totalColumns = new Set();
```

Comment:
for contains and above/below, we'll need to know what columns just came from the current term, separate from the set of all columns that are involved in the query.